Fixed error when using Phantomjs with CORS

Gomes81 opened the pull request:
Fixed error "INVALID_STATE_ERR: INVALID_STATE_ERR: DOM Exception 11" when testing in Phantomjs with CORS active.

Gomes81 committed:
Fixed error when using Phantomjs with CORS

Fixed error "INVALID_STATE_ERR: INVALID_STATE_ERR: DOM Exception 11" when testing in Phantomjs with CORS active.

MauriceButler commented:
Hi @Gomes81 

Can you please explain what the changes you have here actually address?

is it the `apply` ing of an empty array as arguments that triggers something in phantom?

Gomes81 commented:
Hi Maurice,

First I moved ajax.request.withCredentials = !!settings.withCredentials to a line after ajax.request.open, this seems to be an issue in some other browsers too and this seems to be a requirement as documented in Mozilla (here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Requests_with_credentials).

After that I had to remove the bind, since phantomjs tells me that timeout.bind is not defined (and replaced it with and ugly code that use apply).

Hope this explanation helps. Thanks.

MauriceButler commented:
Thanks @Gomes81 

This has been merged in 0aa5750f96500fe50c4936cd9984e1147ad053ea and published as `simple-ajax@2.6.0`
